src/core/qgsmaplayerfactory.h

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+  qgsmaplayerfactory.h
+  --------------------------------------
+  Date                 : March 2021
+  Copyright            : (C) 2021 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPLAYERFACTORY_H
+#define QGSMAPLAYERFACTORY_H
+
+#include "qgis_core.h"
+#include "qgsmaplayer.h"
+
+#include <QString>
+
+/**
+ * \ingroup core
+ * \brief Contains utility functions for creating map layers.
+ *
+ * \since QGIS 3.20
```

Comment:
Assuming you'll backport the PR as a whole, this should be:

```suggestion
 * \since QGIS 3.18.1
```